system-tests/tests/smoke/cre/por_test.go

```diff
@@ -102,6 +108,228 @@ type WorkflowConfig struct {
 	FeedID                 string          `toml:"feed_id" validate:"required,startsnotwith=0x"`
 }
 
+type ReadContractTestInput struct {
+	ExpectedFundingAmount *big.Int
+}
+
+type readContractInput struct {
+	readTargetName       string
+	contractReaderConfig string
+	fundedAddress        string
+	contractName         string
+	contractMethod       string
+}
+
+type registerPoRWorkflowInput struct {
+	WorkflowConfig
+	chainSelector      uint64
+	homeChainSelector  uint64
+	writeTargetName    string
+	workflowDonID      uint32
+	feedID             string
+	addressBook        deployment.AddressBook
+	priceProvider      PriceProvider
+	sethClient         *seth.Client
+	deployerPrivateKey string
+	creCLIAbsPath      string
+	creCLIsettingsFile *os.File
+	authKey            string
+	readContractInput  *readContractInput
+}
+
+type workflowRegistrar struct {
+	configBuilder ConfigBuilderFunc
+}
+
+type ConfigBuilderFunc func(registerPoRWorkflowInput) (*os.File, error)
+
+// create PoR workflow-specific config file
+func buildPoRConfig(input registerPoRWorkflowInput) (*os.File, error) {
+	var secretNameToUse *string
+	if input.authKey != "" {
+		secretNameToUse = ptr.Ptr(AuthorizationKeySecretName)
+	}
+
+	dataFeedsCacheAddress, dataFeedsCacheErr := crecontracts.FindAddressesForChain(input.addressBook, input.chainSelector, df_changeset.DataFeedsCache.String())
+	if dataFeedsCacheErr != nil {
+		return nil, errors.Wrapf(dataFeedsCacheErr, "failed to find data feeds cache address for chain %d", input.chainSelector)
+	}
+
+	workflowConfigFile, configErr := keystoneporcrecli.CreateConfigFile(
+		&libcrecli.PoRWorkflowConfig{
+			FeedID:            input.FeedID,
+			URL:               input.priceProvider.URL(),
+			WriteTargetName:   input.writeTargetName,
+			ConsumerAddress:   dataFeedsCacheAddress.Hex(),
+			AuthKeySecretName: secretNameToUse,
+		},
+	)
+	if configErr != nil {
+		return nil, errors.Wrap(configErr, "failed to create workflow config file")
+	}
+	return workflowConfigFile, nil
+}
+
+// buildReadContractConfig accepts the PoR Workflow Input, deploys a balance reader
```

Comment:
adjust comment? I don't see this function deploying the balance reader, it only searches for it in the address book